DOI.js

```diff
@@ -93,52 +94,83 @@ function completeDOIs(doc) {
 	}
 }
 
-function retrieveDOIs(DOIs, doc) {
-	__num_DOIs = DOIs.length;
 
+var providers = [
+	"11645bd1-0420-45c1-badb-53fb41eeb753", //CrossRef
+	"9f1fb86b-92c8-4db7-b8ee-0b481d456428"  //DataCite
+];
```

Comment:
Don't use globals that get modified during translator execution (via `.shift`) and are not re-initialized when translators are invoked. As per the issue with CSL JSON translator (and EM) above, different invocations of a child translator do not get a fresh sandbox, thus we should assume that every invocation of the translator may have persistent state.

Instead, declare the array in `doWeb` and pass it directly into `retrieveDOIs`.